src/hotspot/share/opto/type.cpp

```diff
@@ -4036,15 +4037,23 @@ int TypeInstPtr::hash(void) const {
 //------------------------------dump2------------------------------------------
 // Dump oop Type
 #ifndef PRODUCT
-void TypeInstPtr::dump2( Dict &d, uint depth, outputStream *st ) const {
+void TypeInstPtr::dump2( Dict &d, uint depth, outputStream* st ) const {
   // Print the name of the klass.
   klass()->print_name_on(st);
 
   switch( _ptr ) {
   case Constant:
-    // TO DO: Make CI print the hex address of the underlying oop.
     if (WizardMode || Verbose) {
-      const_oop()->print_oop(st);
+      st->print(" ");
+      stringStream ss;
+      const_oop()->print_oop(&ss);
+
+      {
+        ResourceMark rm;
```

Comment:
Shouldn't the `ResourceMark` go to before `stringStream ss` which is a `ResourceObj` as well? Also, please add a small comment explaining that this code suppresses the new line emitted by `print_oop`.

Reply:
hi, @TobiHartmann 
Thank you for reviewing this PR. 

stringStream allocates its dynamic buffer using `NEW_C_HEAP_ARRAY`.  IMHO, it's okay without a ResourceMark. 
Unlike `stringStream::grow`, stringStream::as_string(false) does use `NEW_RESOURCE_ARRAY`, which allocates an array on current thread's resource_area. That's why I put ResourceMark in a syntax scope. 

Actually, the current code still works even without that ResourceMark. It's because `Type::dump_on()` has  declared a rm. Let me hoist ResourceMark as you said. That makes code straight-forward locally and I shouldn't assume its context.